libcasr/src/ubsan.rs

```diff
@@ -0,0 +1,151 @@
+//! UndefinedBehaviorSanitizer module implements `Exception` traits for UndefinedBehaviorSanitizer reports.
+use crate::exception::Exception;
+
+use crate::error::*;
+use crate::execution_class::ExecutionClass;
+use regex::Regex;
+
+/// Structure provides an interface for parsing ubsan runtime error message.
+pub struct UbsanException;
+
+impl Exception for UbsanException {
+    fn parse_exception(stderr: &str) -> Option<ExecutionClass> {
+        let stderr_list: Vec<String> = stderr
+            .split('\n')
+            .map(|l| l.trim_end().to_string())
+            .collect();
+        let re = Regex::new(r#".+: runtime error: (.+)SUMMARY: UndefinedBehaviorSanitizer: (\S+)"#)
+            .unwrap();
+        stderr_list
+            .iter()
+            .rev()
+            .find_map(|x| re.captures(x))
+            .map(|cap| {
+                ExecutionClass::new((
+                    "NOT_EXPLOITABLE",
+                    cap.get(2).unwrap().as_str(),
+                    cap.get(1).unwrap().as_str(),
+                    "",
+                ))
+            })
+    }
+}
+
+/// Extract all ubsan responds from stderr
+///
+/// # Arguments
+///
+/// * `stderr_list` - stderr line vector
+///
+/// # Return value
+///
+/// Vector which elements are ubsan reports
+pub fn extract_ubsan_reports(stderr_list: &[&str]) -> Result<Vec<Vec<String>>> {
+    let mut result: Vec<Vec<String>> = vec![];
+    let mut curr: Vec<String> = vec![];
+    let mut in_process = false;
```

Comment:
What is the purpose of this value? Could we get rid of it?

Reply:
No longer relevant